src/io/github/TheBusyBiscuit/GitHubWebAPI4Java/objects/repositories/GitHubRelease.java

```diff
@@ -0,0 +1,135 @@
+package io.github.TheBusyBiscuit.GitHubWebAPI4Java.objects.repositories;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.github.TheBusyBiscuit.GitHubWebAPI4Java.GitHubWebAPI;
+import io.github.TheBusyBiscuit.GitHubWebAPI4Java.annotations.GitHubAccessPoint;
+import io.github.TheBusyBiscuit.GitHubWebAPI4Java.objects.GitHubObject;
+import io.github.TheBusyBiscuit.GitHubWebAPI4Java.objects.users.GitHubUser;
+
+import java.util.Date;
+
+/**
+ * Represents a Release from a GitHubRepository.
+ * @since 1.3.4
+ */
+public class GitHubRelease extends GitHubObject {
+
+    public GitHubRelease(GitHubWebAPI api, GitHubObject parent, int id) {
+        super(api, parent, "/releases/" + id);
+    }
+
+    public GitHubRelease(GitHubWebAPI api, GitHubObject parent, int id, JsonElement response) {
+        super(api, parent, "/releases/" + id);
+
+        this.minimal = response;
+    }
+
+    public GitHubRelease(GitHubObject obj) {
+        super(obj);
+    }
+
+    /**
+     * Returns the ID of this release.
+     * @return the ID of this release.
+     * @throws IllegalAccessException
+     */
+    @GitHubAccessPoint(path = "@id", type = Integer.class, requiresAccessToken = false)
+    public int getID() throws IllegalAccessException {
+        return getInteger("id", false);
+    }
+
+    /**
+     * Returns whether this release is a draft (i.e. not published yet) or not.
+     * @return {@code true} if this release is a draft, {@code false} otherwise.
+     * @throws IllegalAccessException
+     */
+    @GitHubAccessPoint(path = "@draft", type = Boolean.class, requiresAccessToken = false)
+    public boolean isDraft() throws IllegalAccessException {
+        return getBoolean("draft", false);
+    }
+
+    /**
+     * Returns whether this release is indicated as a pre-release or not.
+     * @return {@code true} if this release is a pre-release, {@code false} otherwise.
+     * @throws IllegalAccessException
+     */
+    @GitHubAccessPoint(path = "@prerelease", type = Boolean.class, requiresAccessToken = false)
+    public boolean isPrerelease() throws IllegalAccessException {
+        return getBoolean("prerealease", false);
+    }
+
+    /**
+     * Returns the name of the git tag corresponding to this release.
+     * @return the name of the git tag corresponding to this release.
+     * @throws IllegalAccessException
+     */
+    @GitHubAccessPoint(path = "@tag_name", type = String.class, requiresAccessToken = false)
+    public String getTagName() throws IllegalAccessException {
+        return getString("tag_name", false);
+    }
```

Comment:
Do we have Tag support? If so, we should probably return a Tag object here

Reply:
There's a [GitHubTag](https://github.com/TheBusyBiscuit/GitHubWebAPI4Java/blob/master/src/io/github/TheBusyBiscuit/GitHubWebAPI4Java/objects/repositories/GitHubTag.java) object, but it doesn't look like I can instantiate it from a tag name. I'll investigate that once my rate-limit is lifted.

Reply:
Okay, just checked and I can't instantiate the GitHubTag object based on the provided data ; the Releases API just returns the tag's name whereas the Tags API returns the whole info that is needed.
There's no point making additional calls to the API (getting the list of tags). In my case, I just need the tag name.

If people ever need to get the actual tag object, then they can use the following code:

```java
GitHubRepository repo = (GitHubRepository) parent;
String tagName = getTagName();
        
for (GitHubTag tag : repo.getTags()) {
    if (tag.getName().equals(tagName)) {
        return tag;
    }
}
return null;
```